_src/doc-utils.ts

```diff
@@ -0,0 +1,348 @@
+  const success = await ethereum.request({
+    method: 'wallet_watchAsset',
+    params: {
+      type: parameters.type,
+      options: {
+        address: parameters.address,
+        symbol: parameters.symbol,
+        decimals: parameters.decimals,
+        image: parameters.image,
+      },
+    },
+  });
+
+  if (success) {
+    console.log(`${parameters.symbol} of address ${parameters.address} successfully added to the wallet`);
+  } else {
+    throw new Error(
+      `Something went wrong. ${parameters.symbol} of address ${parameters.address} not added to the wallet`
+    );
+  }
+};
+
+/**
+ * Call this function to make the wallet switch to the desired chain
+ * @param chainId designed chain in HexString
+ * @param ethereum ethereum inpage provider (e.g.: provider loaded by Metamask)
+ */
+const switchToChain = async (chainId: string, ethereum: MetaMaskInpageProvider) => {
+  if (!isChainIdFormatValid(chainId)) {
+    throw new Error(`chainId '${chainId}' must be hexString`);
+  }
+  validateEthereumApi(ethereum);
+  await ethereum.request({
+    method: 'wallet_switchEthereumChain',
+    params: [{ chainId: chainId }],
+  });
+  console.log(`Succesfully switched to chain ${chainId} in metamask`);
+};
+
+/**
+ * Call this function to add a new chain to the wallet. Should only be called if the chain doesn't exist already in the wallet
+ * @param chainId designed chain in HexString
+ * @param ethereum ethereum in page provider (e.g.: provider loaded by Metamask)
+ */
+const addChainToWallet = async (chainId: string, ethereum: MetaMaskInpageProvider) => {
+  if (!isChainIdFormatValid(chainId)) {
+    throw new Error(`chainId '${chainId}' must be hexString`);
+  }
+  validateEthereumApi(ethereum);
+  const chains = await (await fetch(chainsReferenceData)).json();
+  const chain = chains.find((c: any) => toHex(c.chainId) === chainId);
+  if (!chain || !chain.chainId) {
+    console.log(chain);
+    throw new Error(`Chain with chainId '${chainId}' not found in reference data`);
+  }
+
+  const params: AddEthereumChainParameter = {
+    chainId: chainId,
+    chainName: chain.name,
+    nativeCurrency: chain?.nativeCurrency,
+    blockExplorerUrls:
+      chain.explorers && chain.explorers.length > 0 && chain.explorers[0].url
+        ? chain.explorers.map((explorer: any) => explorer.url)
+        : [chain.infoURL],
+    rpcUrls: chain.rpc,
+  };
+
+  const [signerAddress] = (await ethereum.request({ method: 'eth_requestAccounts' })) as string[];
+  await ethereum.request({
+    method: 'wallet_addEthereumChain',
+    params: [params, signerAddress],
+  });
+  console.log(`Chains ${chainId} of params ${JSON.stringify(params)} succesfully added to wallet`);
+};
+
+/**
+ * Functions which validates the format of Link address and interfacs with the contract to make sure
+ * its metadata (e.g.: symbol) is valid
+ * @param address
+ * @param provider
+ */
+const validateLinkAddress = async (address: string, provider: Web3Provider) => {
```

Comment:
This function just validates an address, not a LINK contract address.

Reply:
@obelisk 1st of all thanks for taking the time to review. Actually it does. check the rest of the function:

- 1st it validates the format of an ethereum address by calling `isAddressFormatValid`
- then it connects it loads the contract object `new ethers.Contract` on the right network 
- finally it checks that the symbol contains LINK  and the number of decimals is correct

the goal of this function is to mitigate the risk of typo

Reply:
My mistake, apologies.